azure-pipelines.yml

```diff
@@ -38,41 +38,49 @@ stages:
         echo "##vso[task.setvariable variable=docker_path;]$docker_path"
       displayName: 'Set version number'
 
-    - script: |
-        make setup
+    - script: make setup
       displayName: 'Build & setup'
       env:
         dockerHubUsername: $(dockerHubUsername)
         dockerHubImageName: $(imageName)
-        GOVUK_NOTIFY_API_KEY: $(govukNotifyAPIKey)
+        railsSecretKeyBase: $(railsSecretKeyBase)
+        RAILS_ENV: production
         DEFAULT_PROVIDER_EMAIL: $(defaultProviderEmail)
+        GOVUK_NOTIFY_API_KEY: $(govukNotifyAPIKey)
 
-    - script: |
-        make ci.lint-ruby
+    - script: make ci.lint-ruby
       displayName: 'Rubocop'
       env:
         dockerHubUsername: $(dockerHubUsername)
         dockerHubImageName: $(imageName)
-        GOVUK_NOTIFY_API_KEY: $(govukNotifyAPIKey)
+        railsSecretKeyBase: $(railsSecretKeyBase)
+        RAILS_ENV: production
         DEFAULT_PROVIDER_EMAIL: $(defaultProviderEmail)
+        GOVUK_NOTIFY_API_KEY: $(govukNotifyAPIKey)
 
-    - script: |
-        make ci.lint-erb
+    - script: make ci.lint-erb
       displayName: 'ERB lint'
       env:
         dockerHubUsername: $(dockerHubUsername)
         dockerHubImageName: $(imageName)
-        GOVUK_NOTIFY_API_KEY: $(govukNotifyAPIKey)
+        railsSecretKeyBase: $(railsSecretKeyBase)
+        RAILS_ENV: production
         DEFAULT_PROVIDER_EMAIL: $(defaultProviderEmail)
+        GOVUK_NOTIFY_API_KEY: $(govukNotifyAPIKey)
 
     - script: |
+        docker-compose run --rm web bundle exec rake db:setup
         make ci.test
+        test_result=$?
+        if [ "$test_result" == "0" ]; then true; else cat rspec-results.xml; false; fi
       displayName: 'Execute tests'
       env:
         dockerHubUsername: $(dockerHubUsername)
         dockerHubImageName: $(imageName)
-        GOVUK_NOTIFY_API_KEY: $(govukNotifyAPIKey)
+        railsSecretKeyBase: $(railsSecretKeyBase)
```

Comment:
I think we can lose secretKeyBase if we're not in `RAILS_ENV=production`

Reply:
I can double check, but I'm pretty sure I got an error that the key wasn't present. I only seemed to be able to run it without that key being present in dev mode.

Reply:
Rails should generate a random secret in test and development:

https://github.com/rails/rails/blob/516474def169b38caa743eb44a87e86a12de0595/railties/lib/rails/application.rb#L421-L427

Reply:
Having re-run this in test mode without the rails secret key environment variable it appears to be working. Not sure what was difference when it failed before.